CPP_CODING_STYLE.md

```diff
@@ -0,0 +1,139 @@
+# Gluten CPP Core Guidelines
```

Comment:
Should the file be placed in the top-level folder? We have docs/ to organize documents. Also I'd add to Rui's suggestion that we can set a link from the readme to this file.

Reply:
Also it would be great if we can name the doc file within the same convention with other docs.

Reply:
> Thanks for addressing but can we move it to docs/developers ? Since it's likely that the doc will only be viewed by developers

ok, I will do it

Reply:
> Thanks for addressing but can we move it to docs/developers ? Since it's likely that the doc will only be viewed by developers

done